### PR TITLE
Add mentoring notes for python/palindrome-products

### PR DESCRIPTION
**Pull request by tugrulates**

### CODE REVIEW

File: tracks/python/exercises/palindrome-products/mentoring.md
Comment (kotp) [inline tracks/python/exercises/palindrome-products/mentoring.md:65]: _discussion_ typo but also trailing white space on this line.

File: tracks/python/exercises/palindrome-products/mentoring.md
Comment (kotp) [inline tracks/python/exercises/palindrome-products/mentoring.md:66]: Consider not classifying the mentee.  It may be a common mentee that has mastered the reasonable solution on first try.

On a first or second iteration, or where the author has asked for more... will redirect the attention to the effort, not on some imagined specialness of the author.

File: tracks/python/exercises/palindrome-products/mentoring.md
Comment (kotp) [inline tracks/python/exercises/palindrome-products/mentoring.md:70]: This could benefit from the style of "One Sentence, One Line"

File: tracks/python/exercises/palindrome-products/mentoring.md
Comment (kotp) [inline tracks/python/exercises/palindrome-products/mentoring.md:156]: This line also deviates from the one sentence, one line, style that has been present throughout this document (and is the style we are adopting).

**Review by kotp — CHANGES_REQUESTED** [review]
Suggested change because of typo- and style, while other comments are just thoughts.

File: tracks/python/exercises/palindrome-products/mentoring.md
Comment (tugrulates) [reply tracks/python/exercises/palindrome-products/mentoring.md]: I took a stab at one sentencing these. Let me know if they look off to you still.

File: tracks/python/exercises/palindrome-products/mentoring.md
Comment (tugrulates) [reply tracks/python/exercises/palindrome-products/mentoring.md]: Updated the flow of this section.

The information density is still high, but this can be useful if the discussion goes into time complexity. Let me know if you think I should make it even more brief.

**Comment by tugrulates** [comment]
Thank you for the review @kotp! This is my first mentoring note and I really appreciate you taking the time to guide me.

File: tracks/python/exercises/palindrome-products/mentoring.md
Comment (kotp) [reply tracks/python/exercises/palindrome-products/mentoring.md]: These are mentor notes, so I think those can be a bit dense, but you are
likely to get more feedback from mentors than from students.  I am
saying that letting that go here, we will be able to iterate it on the
longer term.

**Comment by tugrulates** [comment]
I noticed a bad math in the complexity calculation arounf *f*. I will go through it once more and re-publish.

**Comment by tugrulates** [comment]
Bailing out on the exact math since that's better for mentors too. This is what changed since you last saw it:

```
-- Complexity of reasonable solution is **O(*fN²logN*)** or **O(*NlogN*)**
+- Complexity of reasonable solution is slightly higher than **O(*NlogN*)**
-    - for numbers closer to *N²*, *f* will approach ***1/N***
+    - inner loop would take **O(*1/f*)** at first, approaching **O(*1*)** gradually
```

File: tracks/python/exercises/palindrome-products/mentoring.md
Comment (kotp) [inline tracks/python/exercises/palindrome-products/mentoring.md:184]: This line should have the spacing around the delimiter, as it is below.

```suggestion
| Solution | Complexity | Test runtime | Profile runtime | Function calls |
```

**Review by kotp — CHANGES_REQUESTED** [review]
I should have done this in the patch as well, and almost did a correction.  But you got this.

I am going to request a review from someone else as well, but to me this is looking pretty nice.

**Review by kotp — APPROVED** [review]

**Comment by kotp** [comment]
I approve of the changes in the UI based on the review, which is what that asks, but unfortunately, it seems to assume that because I approved of those changes, I approve of the entire pull request.  It is not that I disapprove, but we are still waiting on @BethanyG to review the code part.

So I just wanted to let you know why it has the approval, but is not brought in yet, and to hopefully have someone else that can pull this in pause before going through those steps.

File: tracks/python/exercises/palindrome-products/mentoring.md
Comment (IsaacG) [inline tracks/python/exercises/palindrome-products/mentoring.md:50]: This already is defined for you: https://docs.python.org/3/library/operator.html#operator.ge

File: tracks/python/exercises/palindrome-products/mentoring.md
Comment (IsaacG) [inline tracks/python/exercises/palindrome-products/mentoring.md:62]: "Dealing with correctness" is what the tests, is rather vague, and applies to any exercise equally.

File: tracks/python/exercises/palindrome-products/mentoring.md
Comment (IsaacG) [inline tracks/python/exercises/palindrome-products/mentoring.md:64]: Operators should be space padded.

File: tracks/python/exercises/palindrome-products/mentoring.md
Comment (IsaacG) [inline tracks/python/exercises/palindrome-products/mentoring.md:67]: This point is basically "see more advanced points below"

File: tracks/python/exercises/palindrome-products/mentoring.md
Comment (tugrulates) [reply tracks/python/exercises/palindrome-products/mentoring.md]: This was meant to be math-like notation, but it wasn't the best looking. I now made it more elaborate and ready for mentors to copy paste. Please take another look.

File: tracks/python/exercises/palindrome-products/mentoring.md
Comment (tugrulates) [reply tracks/python/exercises/palindrome-products/mentoring.md]: You are correct. I added it because of the struggle of one author. However, it's a trivial suggestion, I removed it now.

File: tracks/python/exercises/palindrome-products/mentoring.md
Comment (tugrulates) [reply tracks/python/exercises/palindrome-products/mentoring.md]: Could you elaborate a little further? My thought was that it wasn't obvious to the mentor why the rest of the doc exists, so I mentioned when it can be used.

I reworded for further clarity, please review if the new suggestion looks good to you.

**Comment by tugrulates** [comment]
Thank you @IsaacG, these are really helpful!

**Review by BethanyG — APPROVED** [review]

**Pull request merged**
